File: training-5.0/training-5.0/5-3e.py
# https://contest.yandex.ru/contest/59541/problems/E/
n1 = int(input())
s1 = set(map(int, input().split()))
n2 = int(input())
s2 = set(map(int, input().split()))
n3 = int(input())
s3 = set(map(int, input().split()))
res = s1 & s2
merged_s1_s2 = s1.union(s2)
res = res.union(merged_s1_s2 & s3)
print(*sorted(res))

File: training-5.0/training-5.0/5-3e.cpp
// https://contest.yandex.ru/contest/59541/problems/E/
#include <iostream>
#include <vector>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <deque>

using namespace std;

int main() {
	unordered_map<size_t, size_t> nums;
	vector <unordered_set<size_t>> sNums(3);
	size_t n = 0, num = 0;
	for (size_t i = 0; i < 3; i++) {
		cin >> n;
		for (size_t j = 0; j < n; j++) {
			cin >> num;
			sNums[i].insert(num);
		}
	}
	for (size_t i = 0; i < 3; i++) {
		for (size_t num : sNums[i]) {
			nums[num]++;
		}
	}
	deque<size_t> dq;
	for (auto num : nums) {
		if (num.second > 1)
			dq.push_back(num.first);
	}
	sort(dq.begin(), dq.end());
	for (auto s : dq) {
		cout << s << " ";
	}
	cout << endl;
}
